dev/com.ibm.ws.repository.generator/src/com/ibm/ws/repository/generator/SingleJsonRepositoryGenerator.java

```diff
@@ -0,0 +1,254 @@
+package com.ibm.ws.repository.generator;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.apache.tools.ant.BuildException;
+import org.apache.tools.ant.Task;
+
+import com.ibm.ws.repository.common.enums.ResourceType;
+import com.ibm.ws.repository.connections.SingleFileRepositoryConnection;
+import com.ibm.ws.repository.parsers.EsaParser;
+import com.ibm.ws.repository.parsers.Parser;
+import com.ibm.ws.repository.parsers.PaxParser;
+import com.ibm.ws.repository.parsers.ProductParser;
+import com.ibm.ws.repository.parsers.ProductZipParser;
+import com.ibm.ws.repository.parsers.SamplesParser;
+import com.ibm.ws.repository.parsers.ToolParser;
+import com.ibm.ws.repository.resources.writeable.RepositoryResourceWritable;
+import com.ibm.ws.repository.strategies.writeable.AddThenDeleteStrategy;
+
+public class SingleJsonRepositoryGenerator extends Task {
+
+    private File assetFile;
+    private String assetFilePath;
+    private File metadataFile;
+    private String assetType;
+    private ResourceType assetTypeEnumValue;
+    private File jsonFile;
+
+    @Override
+    public void execute() throws BuildException {
+    	
+        // trace input then create MassiveResource.Type enum
+        System.out.println("CreateJsonRepositoryFiles called with following parameters");
+        System.out.println("- AssetFile      = " + getAssetFile().getAbsolutePath());
+        String metadataFileAbsPath = (getMetadataFile() == null) ? "null" : getMetadataFile().getAbsolutePath();
+        System.out.println("- MetadataFile   = " + metadataFileAbsPath);
+        System.out.println("- AssetType      = " + getAssetType());
+        try {
+            assetTypeEnumValue = ResourceType.valueOf(assetType);
+        } catch (IllegalArgumentException iae) {
+            iae.printStackTrace();
+            throw new BuildException(iae);
+        }
+        
+        SingleFileRepositoryConnection mySingleFileRepo = null;
+        if (jsonFile.exists()) {
+        	mySingleFileRepo = new SingleFileRepositoryConnection(jsonFile);
+        } else {
+        	try {
+				mySingleFileRepo = SingleFileRepositoryConnection.createEmptyRepository(jsonFile);
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
```

Comment:
Does this exception need to be handled?

Reply:
Exception is handled in createEmptyRepository.